car-test.mjs

```diff
@@ -60,21 +60,21 @@ async function run () {
         Readable.from(out).pipe(fs.createWriteStream(filename))
       }
 
-      writer.put({ cid: cidA, bytes: bytesA })
+      await writer.put({ cid: cidA, bytes: bytesA })
       console.log('Put cidA', cidA, bytesA)
       await delay(1000)
 
       if (emptyBlock) {
-        writer.put({ cid: cidB, bytes: bytesB })
+        await writer.put({ cid: cidB, bytes: bytesB })
         console.log('Put cidB', cidB, bytesB)
         await delay(1000)
       }
 
-      writer.put({ cid: cidC, bytes: bytesC })
+      await writer.put({ cid: cidC, bytes: bytesC })
       console.log('Put cidC', cidC, bytesC)
       await delay(1000)
 
-      writer.close()
+      await writer.close()
```

Comment:
this `await` is a bit more important, otherwise the timing of the finalisation of the CAR is out of your control

Reply:
This is eerily similar to the fix I proposed here:

https://github.com/web3-storage/ipfs-car/pull/73

It seems to work here, but I think it's sort of cheating to require that the entire car file or stream be written out before reading it. :-)

Reply:
that's true, but in your case you're dealing with a _file_ not a _stream_, if you piped one to the other then you'd have the situation you're talking about, but as far as `CarReader.fromIterable(fs.createReadStream('file'))` is concerned, it's going to chew it up as fast as the filesystem can give it, and if the whole CAR is not already there then bad luck - it'd be like reading all of /var/log/syslog into an editor and complaining that the messages 1 second from now aren't there.

Reply:
and yeah ... the lack of the `await` on the `writer.close()` is a little sus, doesn't mean it won't function properly but at a minimum it's going to do maximum buffering _and_ any errors in the process are going to be uncaught